### ISSUE TITLE
program terminates with sig 6 in yb::tablet::MvccManager::DoGetSafeTime 

### ISSUE THREAD

**Issue by mchang31415** [YugaByte/yugabyte-db#44]
````
"$debugdir:$datadir/auto-load:/usr/bin/mono-gdb.py".

warning: Unable to find libthread_db matching inferior's thread library, thread debugging will not be available.
Core was generated by `/home/yugabyte/tserver/bin/../lib/unwrapped/yb-tserver --flagfile /home/yugabyt'.
Program terminated with signal 6, Aborted.
#0  0x00007fefd273a198 in raise () from /home/yugabyte/tserver/bin/../lib/linuxbrew/libc.so.6
(gdb) bt
#0  0x00007fefd273a198 in raise () from /home/yugabyte/tserver/bin/../lib/linuxbrew/libc.so.6
#1  0x00007fefd273b5ea in abort () from /home/yugabyte/tserver/bin/../lib/linuxbrew/libc.so.6
#2  0x00007fefd67b1afd in yb::(anonymous namespace)::DumpStackTraceAndExit () at ../../../../../src/yb/util/logging.cc:159
#3  0x00007fefd577e04d in google::LogMessage::Fail () at src/logging.cc:1478
#4  0x00007fefd577ff7d in google::LogMessage::SendToLog (this=<optimized out>) at src/logging.cc:1432
#5  0x00007fefd577dbaa in google::LogMessage::Flush (this=this@entry=0x7feac7b974e0) at src/logging.cc:1301
#6  0x00007fefd5780a3f in google::LogMessageFatal::~LogMessageFatal (this=0x7feac7b974e0, __in_chrg=<optimized out>) at src/logging.cc:2013
#7  0x00007fefdc3b4c3c in yb::tablet::MvccManager::DoGetSafeTime (this=this@entry=0xb7f2980, min_allowed=..., deadline=..., max_allowed=...,
    lock=lock@entry=0x7feac7b975a0) at ../../../../../src/yb/tablet/mvcc.cc:196
#8  0x00007fefdc3b5c7f in yb::tablet::MvccManager::UpdatePropagatedSafeTime (this=0xb7f2980, max_allowed=...)
    at ../../../../../src/yb/tablet/mvcc.cc:130
#9  0x00007fefdc37d570 in operator() (__closure=0x56032230) at ../../../../../src/yb/tablet/tablet_peer.cc:236
#10 std::_Function_handler<void(), yb::tablet::TabletPeer::InitTabletPeer(const std::shared_ptr<yb::tablet::enterprise::Tablet>&, const std::shared_future<std::shared_ptr<yb::client::YBClient> >&, const scoped_refptr<yb::server::Clock>&, const std::shared_ptr<yb::rpc::Messenger>&, const scoped_refptr<yb::log::Log>&, const scoped_refptr<yb::MetricEntity>&, yb::ThreadPool*, yb::ThreadPool*)::<lambda()> >::_M_invoke(const std::_Any_data &) (__functor=...) at /home/centos/.linuxbrew-yb-build/Cellar/gcc/5.3.0/include/c++/5.3.0/functional:1871
#11 0x00007fefdae4700d in operator() (this=0x56032230) at /home/centos/.linuxbrew-yb-build/Cellar/gcc/5.3.0/include/c++/5.3.0/functional:2271
#12 yb::consensus::RaftConsensus::UpdateMajorityReplicated (this=0x56032000, majority_replicated_data=..., committed_index=0x7feac7b977c0)
    at ../../../../../src/yb/consensus/raft_consensus.cc:1006
#13 0x00007fefdae29c3e in yb::consensus::PeerMessageQueue::NotifyObserversOfMajorityReplOpChangeTask (this=0x108e4900,
    majority_replicated_data=...) at ../../../../../src/yb/consensus/consensus_queue.cc:1017
#14 0x00007fefd6818334 in operator() (this=<optimized out>)
    at /home/centos/.linuxbrew-yb-build/Cellar/gcc/5.3.0/include/c++/5.3.0/functional:2271
#15 Run (this=<optimized out>) at ../../../../../src/yb/util/threadpool.cc:68
#16 yb::ThreadPool::DispatchThread (this=0x1fc4400, permanent=false) at ../../../../../src/yb/util/threadpool.cc:615
#17 0x00007fefd6814b46 in operator() (this=0x2ddfeb88) at /home/centos/.linuxbrew-yb-build/Cellar/gcc/5.3.0/include/c++/5.3.0/functional:2271
#18 yb::Thread::SuperviseThread (arg=<optimized out>) at ../../../../../src/yb/util/thread.cc:602
#19 0x00007fefd2aad314 in start_thread () from /home/yugabyte/tserver/bin/../lib/linuxbrew/libpthread.so.0
#20 0x00007fefd27eebed in clone () from /home/yugabyte/tserver/bin/../lib/linuxbrew/libc.so.6
(gdb)
```

**Comment by kmuthukk** [comment]
@mbautin - assigning this to you.

Also - we should avoid generating cores in cases such as these, where the code is failing an explicitly coded assertion/invariant check. The log files (*.INFO, *.FATAL) in such as case, typically already have the relevant information (location of assertion failure, and other auxiliary details about the violated invariant).

**Comment by kmuthukk** [comment]
@mbautin - let's close out the issue with a link to the relevant commit.

**Comment by mbautin** [comment]
Fixed in 07ffec98e57ea37de1064034a7f4e20a7a381390.
